pkg-manager/core/src/install/extendInstallOptions.ts

```diff
@@ -144,6 +144,8 @@ export interface StrictInstallOptions {
 
   supportedArchitectures?: SupportedArchitectures
   hoistWorkspacePackages?: boolean
+
+  useBetaCatalogsFeat?: boolean
```

Comment:
Is there a need for a setting? It is opt-in anyway, as the new field in `pnpm-workspace.yaml` should be added.

Reply:
It's mostly to avoid bug reports that catalogs aren't working on the current version 9.0.6. I was thinking it'd be better for it to not work entirely rather than partially work.

Without the setting here, the `catalog:` protocol would work in `pnpm-workspace.yaml`, but we wouldn't replace it on publish, which is the next thing I'm working on.

Alternatively we could:

- Use an environment variable to feature flag this.
- Avoid merging any catalog PRs until it's completely finished.

I thought the temporary argument was the best out of those 3 options, but open to thoughts!